game-core/src/main/java/games/strategy/triplea/ui/MapPanel.java

```diff
@@ -288,21 +288,35 @@ void setUnitHighlight(final Map<Territory, List<Unit>> units) {
     return highlightedUnits;
   }
 
-  public boolean centerOn(final @Nullable Territory territory) {
+  public void centerOn(final @Nullable Territory territory) {
     if (territory == null || uiContext.getLockMap()) {
-      return false;
+      return;
     }
     final Point p = uiContext.getMapData().getCenter(territory);
     // when centering don't want the map to wrap around,
     // eg if centering on hawaii
     super.setTopLeft((int) (p.x - (getScaledWidth() / 2)), (int) (p.y - (getScaledHeight() / 2)));
-    return true;
   }
 
   void highlightTerritory(final Territory territory) {
-    if (centerOn(territory)) {
+    withMapUnlocked(() -> {
+      centerOn(territory);
```

Comment:
I wonder if this changes behaviour.
With the current code, is centerOn ever called with parameters that return false in this method?

Reply:
No, the only calls to `highlightTerritory()` are with a non-`null` reference for `territory`, so, with the Lock Map change, `centerOn()` will always return `true`.  I originally had a `checkNotNull()` precondition in here several PRs ago, but removed it when this method's visibility was reduced to package-private.  I can add it back, if you like.

(BTW, I was the one that recently added a `boolean` return to `centerOn()` to support the Find Territory feature.  It was always a `void` method prior to the addition of Find Territory.)